pandas/core/frame.py

```diff
@@ -11233,6 +11233,9 @@ def corr(
                 f"'{method}' was supplied"
             )
 
+        # clip coefficient to ensure it is within theoretical bounds
+        correl = np.clip(correl, -1, 1)
```

Comment:
It would be better if the `nancorr` function do the clipping in those functions instead of clipping after the fact